src/modules/savedQuery/saga/fetchSavedQuery.ts

```diff
@@ -0,0 +1,34 @@
+import { getContext, put } from 'redux-saga/effects';
+import { SavedQueryAPIResponse } from '../../../types';
+import {
+  KEEN_CLIENT_CONTEXT,
+  NOTIFICATION_MANAGER_CONTEXT,
+} from '../../../constants';
+import { fetchSavedQuery as fetchSavedQueryAction } from '../actions';
+import { savedQueryActions } from '../index';
+import { queriesActions } from '../../queries';
+
+export function* fetchSavedQuery({
```

Comment:
As we discussed - the `fetchSavedQuery` saga should be responsible only for fetching saved query definition.
The missing part is rehydrating the store with saved query name, tags and visualization settings.